src/spaces/discrete.jl

```diff
@@ -273,3 +273,35 @@ function move_agent_single!(
     move_agent!(agent, position, model)
     return agent
 end
+
+"""
+    swap_agents!(agent1, agent2, model::ABM{<:DiscreteSpace})
+
+Swaps agents function used for swapping the postion of two agents.
+"""
+function swap_agents!(agent1, agent2, model::ABM{<:DiscreteSpace})
+    agents_no_pos = []
+
+    pos_a = agent1.pos    
+    pos_b = agent2.pos
+
+    if(isnothing(pos_a))
```

Comment:
The `if(isnothing(` checks and the `agents_no_pos` aren't needed here, since the `pos` field isn't allowed to have a `nothing` value by definition. This part of the Python code can be skipped entirely (not just lines 442-444, but rather all of [437-444](https://github.com/projectmesa/mesa/blob/main/mesa/space.py#L437C2-L444) can be skipped).